### ISSUE TITLE
Error require gem

### ISSUE THREAD

**Issue by tiagonbotelho** [CozyCo/slack-post#8]
Hello, 

I'm trying to use Slack-Post to send messages through Slack and when i require 'slack-post' it keeps giving me a weird error and i cannot understand what is happening with it.

Error:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- slack-post (LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from init.rb:2:in `<main>'

Thank you for your time

**Comment by mgreensmith** [comment]
Hey Tiago, the correct require for this gem is `require 'slack/post'`.

**Comment by jbernardo95** [comment]
I fixed it already, only needs merge: https://github.com/CozyCo/slack-post/pull/9

**Comment by mgreensmith** [comment]
Hi @jbernardo95, I'm not sure why you're proposing that we change the module namespace of this gem. This change would break all existing implementations of this library. Can you explain why `require 'slack/post'` is a problem that needs fixing?

**Comment by tiagonbotelho** [comment]
i know it was but i tried that already and it didnt work i don't know why but it gives out the same error

**Comment by mgreensmith** [comment]
@tiagonbotelho I'm not sure why you're not able to require this gem, but it's not because the module namespace is incorrect. Perhaps it's not installed correctly? (does it show up in the output of `gem list`?)

```
~ $ gem list | grep slack-post
slack-post (0.2.0)
~ $ irb
irb(main):001:0> require 'slack/post'
=> true
irb(main):002:0> Slack::Post
=> Slack::Post
```

**Comment by tiagonbotelho** [comment]
Strange it seems to work in irb but it gives me that weir error when i use it on bundler to work in my project

**Comment by tiagonbotelho** [comment]
require 'slack/post'
require 'date'
 
Slack::Post.configure(  subdomain: 'myslack',
                        token: 'xoxp-3666247738-3666247752-3664405749-809162',
                        username: 'tiagonbotelho')
 
Slack::Post.post 'Ola o meu nome e tiago', '#'

**Comment by mgreensmith** [comment]
@tiagonbotelho 

Ahh, you're using bundler. Becasue the name of this gem does not match the require, you should put it in your Gemfile like this:

```
gem 'slack/post', :require => 'slack/post'
```

**Comment by jbernardo95** [comment]
@mgreensmith That's my point !
Using it with bundler, although I still need to fix I thing I forgot

**Comment by tiagonbotelho** [comment]
# A sample Gemfile
source "https://rubygems.org"

gem 'slack/post', :require => 'slack/post'

**Comment by mgreensmith** [comment]
Sorry @tiagonbotelho I had a typo in my last post.
```
gem 'slack-post', :require => 'slack/post'
```

**Comment by tiagonbotelho** [comment]
same error though sorry about being a pain but it's an important project and this really helps alot(thanks btw)

**Comment by jbernardo95** [comment]
@mgreensmith Check my PR now

**Comment by mgreensmith** [comment]
Guys, I'm sorry, but I don't think I can help you any further. Here's a demonstration of this gem being loaded successfully via Bundler. I don't need to do the require in the Gemfile, but maybe your setup is different (perhaps you're using `Bundler.require` etc.)

```
~/tiago $ cat Gemfile
source 'http://rubygems.org'
gem 'slack-post'

~/tiago $ cat test.rb
require 'slack/post'
puts 'Slack::Post loaded' if defined?(Slack::Post)

~/tiago $ bundle exec ruby test.rb
Slack::Post loaded
```

**Comment by jbernardo95** [comment]
@mgreensmith I got it working both your version and mine
My PR was just to make the import more logic as the name of the gem is `slack-post` and not `slack/post`

**Comment by mgreensmith** [comment]
@jbernardo95 Yes, I agree that your PR would make it simpler to use this gem. Unfortunately, it's already being used in the world by a number of people, and changing module names will break this gem for everyone who's using it. That's not something that I'd be willing to do, I hope you understand.

**Comment by jbernardo95** [comment]
@mgreensmith I do ;)

**Comment by mgreensmith** [comment]
@jbernardo95 A colleague suggested an alternate approach to make it easier to require this gem. I've added a shim file (via https://github.com/CozyCo/slack-post/commit/d965ef365ab7f977db5507533ffa1359b6bef30a), and released version `0.2.1` of the gem. You should now be able to `require 'slack-post'`.

**Comment by jbernardo95** [comment]
@mgreensmith Yep that works two, mine was just a cleaner way
Thanks for your attention ;)
